httphandler/log.go

```diff
@@ -48,6 +51,44 @@ func ScanCSVAccessLogMessage(csvstr string) (AccessMessageData, error) {
 	amd := AccessMessageData{}
 	_, err := fmt.Sscanf(csvstr, "%q, %q, %q, %q, %d, %f, %q", &amd.Method, &amd.Host,
 		&amd.Path, &amd.UserAgent, &amd.StatusCode, &amd.Duration,
-		&amd.RespErr)
+		&amd.RespErr, &amd.Time)
 	return amd, err
 }
+
+// SyncLogMessageData holds all important informations
+// about replication errors
+type SyncLogMessageData struct {
+	Method      string `json:"method"`
+	FailedHost  string `json:"failedhost"`
+	Path        string `json:"path"`
+	SuccessHost string `json:"successhost"`
+	UserAgent   string `json:"useragent"`
+	ErrorMsg    string `json:"error"`
+	Time        string `json:"ts"`
+}
+
+//String produces data in csv format with fields in following order:
+//Method, Host, Path, UserAgent, StatusCode, Duration, RespErr)
+func (slmd SyncLogMessageData) String() string {
+	return fmt.Sprintf("%q, %q, %q, %q, %q, %q",
+		slmd.Method,
+		slmd.FailedHost,
+		slmd.Path,
+		slmd.SuccessHost,
+		slmd.UserAgent,
+		slmd.ErrorMsg)
+}
+
+//JSON produces data in json format with fields in following order
+func (slmd SyncLogMessageData) JSON() ([]byte, error) {
```

Comment:
Maybe you want to implement an [json.Marshaler](https://golang.org/pkg/encoding/json/#Marshaler) interface here?